### PR TITLE
Improve FAH create & delete

### PR DESCRIPTION
**Pull request by inlined**
1. Create now sets deployment-tool so we can track where users are coming from
2. Create now sets a rollout policy like the Console
3. Delete now works
4. Delete no longer makes buildId required

### CODE REVIEW

File: src/commands/apphosting-backends-delete.ts
Comment (bkendall) [inline src/commands/apphosting-backends-delete.ts:49]: Okay, this is odd. According to the approved proposal, the backend ID is supposed to be a required parameter, not a flag. That eliminates the need to pick a backend at all.

File: src/test/init/apphosting/index.spec.ts
Comment (bkendall) [inline src/test/init/apphosting/index.spec.ts:73]: Nothing here tests to make sure the rollout policy was called. Please add the appropriate logic here to test for that

**Review by bkendall — CHANGES_REQUESTED** [review]

File: src/commands/apphosting-backends-delete.ts
Comment (inlined) [reply src/commands/apphosting-backends-delete.ts:49]: Irrespective of the proposal, do you like this better? It seemed odd to me that location could be browsed but backend couldn't. If you like this better we can push it through API council. As an EAP API, do you feel like it is OK to release and then get approval, or should I splice this into another CL until after approval?

File: src/test/init/apphosting/index.spec.ts
Comment (inlined) [reply src/test/init/apphosting/index.spec.ts:73]: I may need to revert the label. I originally wrote the test to exercise this but realized it was for a smaller utility function and the larger flow is completely untested. I can either:

1. Move creation of the rollout policy into a utility function and test it. This would be a trivial whitebox test that is unlikely to catch anything though.
2. Add a new test that covers the overall flow of the entiere init process.
3. Cry that init functions are generally sparsely tested in this codebase.

You make the call and I'll follow.

File: src/commands/apphosting-backends-delete.ts
Comment (bkendall) [reply src/commands/apphosting-backends-delete.ts:49]: I personally don't like it better, because I hate the prompt selector flow in a CLI 🤷‍♂️ It's not a wrong thing to do, but I'm assuming that there were product discussions that brought us to the API proposal (as that's a requirement at the top of the document) and that the pattern in the proposal was agreed upon by all the relevant stakeholders already. If you'd like to make a change to what was proposed there, starting back in the process, involving the right people, and getting an addendum would be appropriate.

That said, I thought the general pattern was to:

1. require backendID as a parameter in the command
2. default to looking at us-central1 as we 'traditionally' have

If someone is outside of us-central1, they get to include the `--location` flag.

Having this flow also makes it more complex. Sticking with the picker, you have to also consider what happens in non-interactive environment (like a CI system). As written, this would hang, I believe, which may be unexpected (instead of simply failing out).

File: src/test/init/apphosting/index.spec.ts
Comment (bkendall) [reply src/test/init/apphosting/index.spec.ts:73]: Ah, yeah, this isn't actually testing `doSetup`. Probably just revert the test name then. init function coverage is kinda a known weak point.

**Review by bkendall — CHANGES_REQUESTED** [review]

File: src/commands/apphosting-backends-delete.ts
Comment (inlined) [reply src/commands/apphosting-backends-delete.ts]: (If it's chosen that we should keep the existing dialog, I agree that we need to check for nonInteractive)

**Comment by inlined** [comment]
Controversial UX changes have been removed and test name has been reverted. PTAL.

File: src/gcp/apphosting.ts
Comment (bkendall) [inline src/gcp/apphosting.ts:201]: Yes, it should be.

**Review by bkendall — APPROVED** [review]

**Pull request merged**
